channeldb/db.go

```diff
@@ -136,6 +137,13 @@ var (
 			number:    13,
 			migration: migration13.MigrateMPP,
 		},
+		{
+			// Initialize payment address index and begin using it
+			// as the default index, falling back to payment hash
+			// index.
+			number:    14,
+			migration: mig.CreateTLB(payAddrIndexBucket),
```

Comment:
Are we sure we don't want to go through the existing invoices and fill the index in the migration?

Reply:
we wouldn't be able to do all them because really old invoices have a pay addr of all 0's. we'd only be able to do so for invoices created post-0.8, but i'm not sure the extra work is worth it considering we have the fallback. this is definitely debatable tho

Reply:
Yes it would just be to not have invoices in the database that have a payment addr (post 0.8) but are not in the index.

Reply:
Interesting migration variant. Is this needed if `CreateBucketIfNotExists` is used everywhere? Or perhaps this saves on nil checks as we can assume the bucket is always there?

Reply:
We need to bump the version anyway, otherwise people may rollback and our invoice accounting could be incorrect. There will also be another PR adding the set_id index which will need the same.

If we can avoid it, I would prefer not using `CreateBucketIfNotExists` for static buckets that _should_ be there, and save it for buckets whose name isn't known at compile time. I added this migration to make that process easier for us going forward.
 - 1 LOC vs 4
 - fewer untested codepaths/improved coverage
 - slightly more performant
 - more clear which buckets can change and which are assumed to be part of the schema

imo these bucketing changes should be added as a migration unless it is truly optional and we want to allow users to downgrade, but i don't see that as being very likely since I can't recall the last release without a migration

Reply:
I tend to agree with avoiding `CreateBucketIfNotExists`. On the other hand, if we are moving away from migrations in some cases in favor of optional buckets ('on the fly migrations'), then `CreateBucketIfNotExists` is just an instance of that. If we do optional buckets for on-the-fly migrations, why not keep it for the root buckets?

One downside of this migration is that it isn't easily possible to downgrade for example to the latest tagged release during development.

Reply:
Discussed offline. Downgrade isn't safe anyway. We probably also need to bump the version for other on-the-fly migrations